File: backend/models/Post.mjs
import mongoose from "mongoose";

const PostSchema = new mongoose.Schema({
        title: {
            type: String,
        },
        textInfo: {
            type: String,
        },
        user: {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User",
            required: true,
        },
        imageUrl: String,
    }, {
        timestamps: true,
    },
);

export default mongoose.model('Post', PostSchema);

File: mobile/cardiowell/lib/models/post.dart
class Post {
  final String title;
  final String textInfo;
  final String user;
  final String id;
  final String imageUrl;
  final DateTime createdAt;
  final DateTime updatedAt;

  Post({
    required this.title,
    required this.imageUrl,
    required this.textInfo,
    required this.user,
    required this.id,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    final userJson = json['user'] as Map<String, dynamic>;

    return Post(
      title: json['title'],
      textInfo: json['textInfo'],
      imageUrl: json['imageUrl'],
      user: userJson['_id'],
      id: json['_id'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }
}
